src/uipath_langchain/tools/preconfigured.py

```diff
@@ -0,0 +1,178 @@
+import json
+import logging
+from typing import Iterable, Optional, Type
+
+import requests
+from jsonschema_pydantic import jsonschema_to_pydantic as create_model  # type: ignore
+from langchain_community.cache import SQLiteCache
+from langchain_core.tools import BaseTool, StructuredTool
+from langgraph.types import interrupt
+from pydantic import BaseModel
+from uipath import UiPath
+from uipath.agent.models.agent import (
+    AgentDefinition,
+    AgentEscalationChannel,
+    AgentEscalationResourceConfig,
+    AgentIntegrationToolResourceConfig,
+    AgentProcessToolResourceConfig,
+    AgentResourceConfig,
+)
+from uipath.models import CreateAction, InvokeProcess
+
+
+def create_process_tool(resource: AgentProcessToolResourceConfig) -> Iterable[BaseTool]:
+    async def process(**kwargs) -> BaseModel:
+        return interrupt(
+            InvokeProcess(
+                name=resource.name,
+                input_arguments=kwargs,
+                process_folder_path="Shared/A2ALoanCreditRatingTool",
+            )
+        )
+
+    input_schema = create_model(resource.input_schema)
+
+    class ProcessTool(StructuredTool):
+        OutputType: Type[BaseModel] = create_model(resource.output_schema)
+
+    yield ProcessTool(
+        name=resource.name,
+        args_schema=input_schema,
+        description=resource.description,
+        coroutine=process,
+    )
+
+
+def create_escalation_tool_from_channel(channel: AgentEscalationChannel) -> BaseTool:
+    async def escalate(**kwargs) -> BaseModel:
+        return interrupt(
+            CreateAction(
+                title=channel.description,
+                data=kwargs,
+                assignee=None,  # It does not make sense to expose the recipient choice to the agent even though it's technically possible.
+                app_name=channel.properties.app_name,
+                app_folder_path=channel.properties.folder_name,
```

Comment:
folder name is not the folder path (the folder path represents the friendly name used to identify a folder and it contains the whole folder hierarchy)

Reply:
Channels do not specify folder paths. I've changed this to `None`. Please resolve this conversation if that assumption is sufficient.